File: html/ledtest.php
<!DOCTYPE html>
<html>
	<head>
		<title>CaravanPi Konfiguration</title>
		
		<!-- zu verwendender Zeichensatz -->
		<meta http-equiv="Content-type" content="text/html; charset=utf-8" />
		
		<!-- Stylesheets -->
		<link rel="stylesheet" type="text/css" href="css/main.css">
		<link rel="stylesheet" type="text/css" href="css/custom.css">
		
		<!-- Favicon - Icon im Browser-Tab -->
		<link rel="icon" href="icons/CaravanPi.ico">
		
		<!-- Icons -->
		<link rel="apple-touch-icon" href="modules/MMM-Remote-Control/apple-touch-icon.png">
		<meta name="apple-mobile-web-app-title" content="MagicRemote">
		<meta name="apple-mobile-web-app-capable" content="yes">
		<meta name="mobile-web-app-capable" content="yes">
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
		
		<!-- Anweisung für Darstellung auf mobilen Devices (SmartPhone, Tablet) -->
		<meta name="viewport" content="width=device-width, initial-scale=1">
		
		<!-- Anweisung für Goolge: Seite nicht übersetzen -->
		<meta name="google" content="notranslate" />
	</head>

	<body>
		<header class="header">CaravanPi Konfiguration - LEDs testen</header>
	
		<div class="container">
			<form class="caravanpi-form" action="ledtest.py" method="post">

				<!-- Leergewicht -->
				<div class="row">
					<div class="col-75">
						<label for="color">LED auf ff. Farbe schalten: </label>  
					</div>
					<div class="col-25">
					  <select name="color" size="1">
               <option value="2">blau</option>
              <option value="1">blau blinkend</option>
             <option selected value="0">rot</option>
              <option value="-1">grün blinkend</option>
              <option value="-2">grün</option>
            </select>
					</div>
				</div>
        
					<!-- Button -->
				<div class="row submit-button">
					<input type="submit" id="submit" name="submit" value="LEDs für 60 Sekunden einschalten">
				</div>
			</form>
		</div>
	
		<div id="power-button" class="menu-element hidden button main-menu">
			<a class="text" href="index.html">zurück zum Konfigurationsmenü</a>
		</div>

	</body>
</html>

File: html/ledtest.py
#!/usr/bin/python3
# coding=utf-8
# ledtest.py
#
# startet LED Test Routinen
#
#-------------------------------------------------------------------------------

import time, datetime
import signal
import sys
from time import sleep

# CGI handling
import cgi
import cgitb

# import for opening URL to change update Intervall on magic Mirror
import urllib.request

# -----------------------------------------------
# libraries from CaravanPi
# -----------------------------------------------
sys.path.append('/home/pi/CaravanPi/.lib')
from CaravanPiFilesClass import CaravanPiFiles
from CaravanPiFunctionsClass import CaravanPiFunctions

# -----------------------------------------------
# global variables
# -----------------------------------------------

def main():
	# -------------------------
	# main 
	# -------------------------

	# lesen der Werte aus der HTML form
	cgitb.enable(display=0, logdir="/var/log/apache2")
	
	form = cgi.FieldStorage()
	
	cgi_color = form.getvalue('color')
	
	#print(cgi_color)
	
	if (cgi_color != None):
		CaravanPiFiles.writeTestColor(0, 0, cgi_color)
		# send kill SIGUSR2 to position2file.py
		urllib.request.urlopen('http://127.0.0.1:8089/ConfigSite/LEDtest')
		if cgi_color == "-2":
			strColor = "grün"
		elif cgi_color == "-1":
			strColor = "grün blinkend"
		elif cgi_color == "0":
			strColor = "rot"
		elif cgi_color == "1":
			strColor = "blau blinkend"
		elif cgi_color == "2":
			strColor = "blau"
		elif cgi_color == "99":
			strColor = "aus"
		else:
			strColor = "----"

	# Ergebnis Website schreiben
	print("Content-Type: text/html; charset=utf-8\n\n")
	print("<html>")
	
	print("<head>")
	print("<title>CaravanPi Konfiguration</title>")
	print("<meta http-equiv='refresh' content='2; URL=ledtest.php'>")
	print("<link rel='stylesheet' type='text/css' href='css/main.css'>")
	print("<link rel='stylesheet' type='text/css' href='css/custom.css'>")
	print("</head>")

	print("<body>")
	print('<header class="header">CaravanPi Konfiguration - LED Test</header>')
	if (cgi_color != None):
		print("Die Testroutine wurde gestartet. Die LEDs schalten für mindestens 60 Sekunden auf "+strColor)
	else:
		print("ES KONNTEN KEINE WERTE AUS DEM FORMULAR GELESEN WERDEN!")
	print("<br/><br/>Sie werden zur Eingabeseite weitergeleitet")
	print("</body>")
	print("</html>")


if __name__ == "__main__":
	main()
